projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -370,6 +370,33 @@ def is_vectorizer_disabled_query(self) -> sql.Composed:
             self.vectorizer_table_ident,
         )
 
+    @cached_property
+    def requeue_or_remove_work_query(self) -> sql.Composed:
+        return sql.SQL("""
+            MERGE INTO {queue_table} AS target
+            USING (
+                SELECT
+                    id
+                FROM
+                    {queue_table}
+                WHERE
+                    id = {vectorizer_id}) AS source ON target.id = source.id
+            WHEN MATCHED
+                AND target.retries >= %s THEN
+                    DELETE
+            WHEN MATCHED THEN
```

Comment:
What about adding an extra check so the query won't ever remove a row if there is a backoff in place?
```suggestion
            WHEN MATCHED
                AND retry_after <= now() THEN
```

Reply:
I just wrote a test for this case, it's already working, as this is handled in the fetch_work_query func:

```sql
                WITH selected_rows AS (
                    SELECT {pk_fields}
                    FROM {queue_table}
                    WHERE retry_after is null or retry_after < now()
                    LIMIT %s
                    FOR UPDATE SKIP LOCKED
                )
```

Reply:
just pasted the initial part of the query not to make it not to pollute the conversation

Reply:
I still believe it is part of this query to protect itself against an unexpected behaviour